### PR TITLE
Fix our MSB3270 warnings

### PR DESCRIPTION
**Pull request by jaredpar**
There are a few parts to this change:

1. Ensure that MSB3270 and MSB3277 warnings are promoted to errors in CI
and hence block merging.
1. Move our DiaSymReader.Native logic into a separate targets file.
This resource cannot be consumed with a simple package reference but
rather requires a package reference and custom logic to pull out the
contained binaries. This logic used to be spread through our build. Now
it's in a single place.
1. Remove the x86 bootstrapping logic. This was testing a pretty obscure
scenario and the cost of maintaining that logic is siginificant at this
point. Can bring back if we ever find a bug in this area.

The root cause of the MSB3270 warnings is a subtle change in the SDK. It
now passes the runtime graph to NuGet for native assets. In the case of
DiaSymReader.Native there are runtime specific assets hence the SDK /
NuGet had to pick one for framework projects. This eventually lead to
`PlatformTarget` being set to x86 where it shoud have been AnyCPU.

Part of the change includes adding `ExcludeAssets=all` to the package
reference which means they no longer figure into this logic and hence
the binaries are marked as AnyCPU.

This regression in behavior is being tracked by
https://github.com/dotnet/sdk/issues/3495

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-infrastructure , @nguerrera for review

File: eng/targets/Imports.targets [hunk 1]
```diff
@@ -18,12 +18,6 @@
       which has all of our global settings 
     -->
     <UserRuntimeConfig Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(OutputType)' == 'Exe'">$(RepositoryEngineeringDir)config\runtimeconfig.template.json</UserRuntimeConfig>
-
-    <!-- 
-      Bootstrap compiler should run as a 32 bit process on our 32 bit test runs. This will validate
-      the legacy JIT ability to handle the compiler. 
-    -->
-    <PlatformTarget Condition="'$(OutputType)' == 'Exe' AND '$(TargetFramework)' == 'net472' AND '$(BOOTSTRAP32)' == 'true'">x86</PlatformTarget>
```
Comment (333fred) [inline eng/targets/Imports.targets]: This was specifically introduced after a legacy jit regression issue. What are we doing to replicate the test?

File: eng/targets/DiaSymReaderNative.targets [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<Project>
```
Comment (tmat) [inline eng/targets/DiaSymReaderNative.targets]: Nit: Missing copyright header.

I'd add a comment that describes what the file is for.

File: eng/targets/DiaSymReaderNative.targets [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  <PropertyGroup>
+    <PlatformTarget Condition="'$(PlatformTarget)' == ''">AnyCPU</PlatformTarget>
+  </PropertyGroup>
```
Comment (tmat) [inline eng/targets/DiaSymReaderNative.targets]: Is this related to DSRN?

File: eng/targets/Imports.targets (hunk 1 above)
Comment (jaredpar) [reply eng/targets/Imports.targets]: This was done to guard against a potential legacy JIT issue. Eventually we determined that it was just a more vanilla race condition. At the time though we were quite stumped and decided to do this as a test. The compiler is fairly well tested on the legacy JIT via our unit tests including minimal execution of csc.exe. Hence the added coverage here is very minimal and not worth the infrastructure to maintain.

File: eng/targets/DiaSymReaderNative.targets (hunk 2 above)
Comment (jaredpar) [reply eng/targets/DiaSymReaderNative.targets]: Added both. Looks like we missed copyrights in some other files so added those as well.

File: eng/targets/DiaSymReaderNative.targets (hunk 3 above)
Comment (jaredpar) [reply eng/targets/DiaSymReaderNative.targets]: Not sure what DSRN but added a comment + linked issue to describe the reasoning for this.

**Comment by nguerrera** [comment]
I see this is going into a 16.4 branch. I think if you build 16.3 with a recent enough sdk but no so recent that it has the (non-existent yet fix for https://github.com/dotnet/sdk/issues/3495), you could end up with the exes being x86. Are we sure the toolset won't be updated to such a build from here for 16.3 branches?

Maybe at least port the warning as error part to 16.3 as a safeguard?

File: src/Compilers/CSharp/csc/csc.csproj [hunk 4]
```diff
@@ -42,4 +41,5 @@
     </None>
   </ItemGroup>
   <Import Project="..\..\Core\CommandLine\CommandLine.projitems" Label="Shared" />
+  <Import Project="$(RepositoryEngineeringDir)targets\DiaSymReaderNative.targets" />
```
Comment (sharwell) [inline src/Compilers/CSharp/csc/csc.csproj]: ❔ Does this resolve before Arcade is restored? Want to make sure it doesn't block project load when opening before a command line restore.

File: eng/targets/DiaSymReaderNative.targets [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  <PropertyGroup>
+    <!-- 
+      Ensure the binaries using DiaSymReader.Native are marked as AnyCPU unless they specify 
+      a different target. This should be the default behavior but recent SDK changes to more 
+      correctly consider RIDs caused our behavior to change here. Once the SDK logic is settled
+      here we can remove this.
+
+      https://github.com/dotnet/sdk/issues/3495
+    -->
+    <PlatformTarget Condition="'$(PlatformTarget)' == ''">AnyCPU</PlatformTarget>
```
Comment (sharwell) [inline eng/targets/DiaSymReaderNative.targets]: ❔ Have we verified that this occurs before whatever operation was causing the SDK to set this property to x86?

File: eng/build-utils.ps1 [hunk 6]
```diff
@@ -279,7 +279,7 @@ function Make-BootstrapBuild([switch]$force32 = $false) {
   $projectPath = "src\NuGet\$packageName\$packageName.Package.csproj"
   $force32Flag = if ($force32) { " /p:BOOTSTRAP32=true" } else { "" }
 
-  Run-MSBuild $projectPath "/restore /t:Pack /p:DotNetUseShippingVersions=true /p:InitialDefineConstants=BOOTSTRAP /p:PackageOutputPath=`"$dir`" /p:EnableNgenOptimization=false $force32Flag" -logFileName "Bootstrap" -configuration $bootstrapConfiguration
+  Run-MSBuild $projectPath "/restore /t:Pack /p:RoslynEnforceCodeStyle=false /p:UseRoslynAnalyzers=false /p:DotNetUseShippingVersions=true /p:InitialDefineConstants=BOOTSTRAP /p:PackageOutputPath=`"$dir`" /p:EnableNgenOptimization=false $force32Flag" -logFileName "Bootstrap" -configuration $bootstrapConfiguration
```
Comment (sharwell) [inline eng/build-utils.ps1]: > /p:UseRoslynAnalyzers=false

Are we sure we want to make this change? @mavasani have we ever caught bugs by running analyzers in the bootstrap build?

File: eng/targets/DiaSymReaderNative.targets (hunk 5 above)
Comment (jaredpar) [reply eng/targets/DiaSymReaderNative.targets]: Yes.

**Comment by jaredpar** [comment]
@nguerrera 

> Maybe at least port the warning as error part to 16.3 as a safeguard?

The warning exists in 16.3 so can't make it an error without also fixing it.

File: eng/build-utils.ps1 (hunk 6 above)
Comment (jaredpar) [reply eng/build-utils.ps1]: 100% sure. All of this code is immediately built again with analyzers enabled. Running analyzers during bootstrap build is just burning CPU cycles and making it harder to turn on warn as error. Which is precisely how we got into this state to begin with.

File: src/Compilers/CSharp/csc/csc.csproj (hunk 4 above)
Comment (jaredpar) [reply src/Compilers/CSharp/csc/csc.csproj]: This style of Import is used already in this project hence it should be no different than the behavior we have today.

**Comment by jaredpar** [comment]
Had offline discussion. Going to merge this here to stop the bleeding and then port back to master. Because it's an infra only change and doesn't affect the produced binaries it doesn't need to go through ask / tell mode.
